[Done]Added student form UI for name and date.

cocoral opened the pull request:
To avoid huge pr... just create one here for now. Main change: 
Added Student name and date onto student form page.

TODO: 
* Hook up redux-form
* Consolidate the name and datepicker component so student and company form can share them. 

![image](https://user-images.githubusercontent.com/16631140/54066775-14eeaf80-4204-11e9-950e-c969d5abe1ab.png)

cocoral committed:
Added student form UI for name and date.

Haven't hook up redux-form yet!

cocoral committed:
Merge remote-tracking branch 'origin/develop' into feature/student-form

susiekims commented on src/app/components/Student.js:
```diff
@@ -1,5 +1,56 @@
-const Student = () => <h1>Student</h1>;
+const Student = () => {
+  const [selectedDate, handleDateChange] = useState(new Date());
+  const [name, handleName] = useState('');
```
i love it!!! :anchor:

skrelab replied:
OOOooo me too! Never used hooks before

cocoral replied:
It's gone when redux form is in lol. All states are handled!

susiekims commented on src/app/components/Student.js:
```diff
@@ -1,5 +1,56 @@
+  return (
+    <Grid container justify="center">
+      <Grid item xs={10} sm={8} md={6}>
+        <div style={{ padding: 50 }}>
```
I think we can remove this padding so that all the pages will have the same width :) i know this is just a WIP so we can always change this later

cocoral replied:
I didn't dismiss this! It has been updated in the file :)

susiekims reviewed (dismissed):
Nice work! love the hooks 😍

cocoral committed:
Merge remote-tracking branch 'origin/develop' into feature/student-form

cocoral committed:
Hooked redux-form with student form

1. connect student name and datepicker with redux form.
2. Updated name and datepicker component so they can be shared by tow forms
3. dispatch a `POST_STUDENT_SUCCESS` when submit valid button (but it's actually not posting)

cocoral committed:
made validation for student name working

cocoral committed:
Make student component a container.

cocoral committed:
removed padding

cocoral committed:
Merge remote-tracking branch 'origin/develop' into feature/student-form

cocoral committed:
Added an id on student name field

cocoral commented:
Big PR warning! 

* Connected student name and datepicker with redux form.
* Updated name and datepicker component so they can be shared by two forms
** renamed CompanyName component to NameField, and the `name` is used to differentiate the fields ;
** InterviewDate component stayed as it is
* Dispatched a `POST_STUDENT_SUCCESS` when submit valid button (but it's actually not posting)

The flow is basically the same with company from, and style is matching company form as well

cocoral committed:
Solved a warning.. and realized that student form doesn't need to be a container.

"Warning: Failed prop type: Invalid prop `component` of type `object` supplied to `Route`, expected `function`."

susiekims commented on src/app/components/StudentForm.js:
```diff
@@ -0,0 +1,59 @@
+  initialValues: {
+    interviewDate: new Date()
+  }
+})(connect(mapStateToProps)(Student));
```
double curry!!!! 😮 👏

cocoral replied:
Too much ()

susiekims reviewed (approved):
bomb.com! 💣 🙌

cocoral commented:
Re: last commit 
When I pass the student form component into the route file, it threw this warning in console: 

```
Warning: Failed prop type: Invalid prop `component` of type `object` supplied to `Route`, expected `function`.
```

To solve this, I extract the student form out and placed it into a pure functional component. 
Looks like it's react router do not like the output from `reduxForm({someAttributes})(TheForm)`?
